bench/irmin-pack/tree.ml

```diff
@@ -54,19 +103,93 @@ module Generate_trees_from_trace
 struct
   type t = { mutable tree : Store.tree }
 
-  type stats = {
-    mutable adds : int;
-    mutable removes : int;
-    mutable finds : int;
-    mutable mems : int;
-    mutable mem_tree : int;
-    mutable copies : int;
-  }
-
-  let stats =
-    { adds = 0; removes = 0; finds = 0; mems = 0; mem_tree = 0; copies = 0 }
-
-  let get_stats () = stats
+  type stat_entry =
+    [ `Add | `Remove | `Find | `Mem | `Mem_tree | `Checkout | `Copy | `Commit ]
+  [@@deriving repr]
+
+  let op_tags =
+    [ `Add; `Remove; `Find; `Mem; `Mem_tree; `Checkout; `Copy; `Commit ]
+
+  (** One running histogram for each trace base operartions.
+
+      The histograms are computed using https://github.com/barko/bentov.
+
+      [Bentov] does exactly the right thing here, i.e. computing dynamic
+      histograms without the need for a priori information on the distributions,
+      while maintaining a constant memory space and a marginal CPU footprint.
+
+      The implementation of that library is pretty straightforward, but not
+      perfect; it doesn't scale well with the number of bins. I chose 32
+      randomly.
+
+      The computed histogram depends on the order of the operations, some
+      maginal unsabilities are to be expected. *)
+  let histo_per_op =
+    op_tags
+    |> List.map (fun which -> (which, Bentov.create 32))
+    |> List.to_seq
+    |> Hashtbl.of_seq
+
+  (** Find and print the largest directory. This may be useful in the future to
+      make sure that we are benching on million-sized directories. *)
```

Comment:
I think we should not add code that we don't use right now. If you think it's useful you can add a cli argument to print this?

Reply:
I'd have to refactor a big chunk of code to avoid printing that in the middle of the progress bar. My next PR on that file will most likely take care of it, so I'd like to keep it if that is OK for you.